pkg/controller/backup_controller.go

```diff
@@ -281,6 +290,47 @@ func (c *backupController) getLocationAndValidate(itm *api.Backup, defaultBackup
 	return backupLocation, validationErrors
 }
 
+// getVolumeLocationAndValidate ensures:
+// - each VSL name in Spec VolumeSnapshotLocation exists as a VSL
+// - exactly 1 VSL per existing or default provider
+// - a given default provider's VSL name is added to the Spec VolumeSnapshotLocation if it does not exist as a VSL
+func (c *backupController) getVolumeLocationAndValidate(itm *api.Backup, defaultVSLs map[provider]string) []string {
```

Comment:
maybe `defaultAndValidateSnapshotLocations`? It's a long one I know but i think that's a better description of what it does